packages/app/utils/transaction.ts

```diff
@@ -11,11 +13,20 @@ import {
 import { Networks } from '@walless/core';
 import { db } from 'utils/storage';
 
-import { connection } from './../../network/src/utils/connection';
-
 const solConn = new Connection(clusterApiUrl('devnet'));
 const sampleKeypair = Keypair.generate();
 
+// Use this connection to get metadata
+const suiConn = new SuiConnection({
+	fullnode: 'https://fullnode.devnet.sui.io:443/',
+});
```

Comment:
this suiConn object should be resued from one single source, could be singleton pattern.. for cases like we have network switch UI -- this will be change accordingly to the selected network (by the user).

Reply:
Will I temporary import conn from web/scripts/kernel which has exported connection, but they are in different context? We may need communication module for user settings. At the moment, these connection is reused, but it should be initialized dynamicly base on user settings